### DISCUSSION TITLE
listen to the evt of targetArrowheadTool

### DISCUSSION THREAD

**Discussion by Ndmanh123** [clientIO/joint#1682] [Q&A] [answered]
Hi  i have use targetArrowheadTool so that i can move link from a cell to another but i cant find the way to get the information about the target of the link before and after the change . If some one have a answer let me know thank

**Comment by jamesgeorgewilliams** [comment, marked as answer]
Hi,

you could try something like the following:

It is preferential to listen to `link:connect` and `link:disconnect` on the `paper`.

If you listen to events on the `graph`, you will get all the intermediate events as the link follows the cursor which probably isn't desirable in this instance.

The example below shows how to get both previous and next model in a single function `onReconnect`. 

```js
function onReconnect(prevModel, nextModel, endType) {
    console.log(`${endType} changed from ${prevModel?.id} to ${nextModel?.id} `);
}

paper.on({
    'link:disconnect': (linkView, evt, prevEndView, prevEndMagnet, endType) => {
        evt.data.previous = prevEndView.model;
        if (!linkView.model.prop([endType, 'id'])) {
            onReconnect(prevEndView.model, null, endType);
        }
    },
    'link:connect': (linkView, evt, newEndView, newEndMagnet, endType) => {
        const { previous = null } = evt.data;
        onReconnect(previous, newEndView.model, endType);
    },
});
```

---

Listening to events on `graph` example, but maybe not desirable in this case. As mentioned above, it would be better to listen to `link:connect` and `link:disconnect` on the `paper`.

```js
const link = new joint.dia.Link();
link.source(rect);
link.target(rect2);
link.addTo(graph);

const targetId = link.target().id;
const targetElementBefore = graph.getCell(targetId); 

graph.on('change:source change:target', function(link) {
    const newTargetId = link.target().id;
    const targetElementAfter = graph.getCell(newTargetId);
});
```

https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#dia.Graph.prototype.getCell
https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#dia.Link.events

Thanks,

James

**Reply by babyandy0111** [reply]
Hi James 
How can i get  my select cell from targetElement ?

**Reply by jamesgeorgewilliams** [reply]
@babyandy0111 Hi, can you create a new discussion, and explain your use-case? I am not totally sure what you mean.  Thanks
